File: 1erdepartamental/python/2.suma_parametros/main.py
"""
Programa: Suma con Parámetros
Demuestra parámetros por defecto y operator overloading
"""

from core.sumador import Sumador

if __name__ == "__main__":
    s1 = Sumador(10, 20)
    print(f"s1: {s1}")
    print(f"Suma: {s1.sumar()}\n")

    s2 = Sumador(5, 15)
    s3 = s1 + s2
    print(f"s1 + s2 = {s3}")

File: 1erdepartamental/c++/2.suma_parametros/main.cpp
#include <iostream>
#include "core/Sumador.h"

int main() {
    // Crea primer sumador
    Sumador s1(10, 20);
    std::cout << "s1: " << s1 << std::endl;
    std::cout << "Suma: " << s1.sumar() << std::endl << std::endl;
    
    // Crea segundo sumador y suma con el primero
    Sumador s2(5, 15);
    // Usa el operador + sobrecargado
    Sumador s3 = s1 + s2;
    std::cout << "s1 + s2 = " << s3 << std::endl;
    
    return 0;
}
